scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -969,7 +998,9 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
       if (advance) next()
       res
     case token: Unquote =>
-      unquote[T]
+      unquote[T](advance)
```

Comment:
In old form, `unquote` always drives iterator forward. However, sometimes that is not desired behaviour, so I added extra parameter with default value, for the future cases.